addons/EmpyPopTracker/EmpyPopTracker.lua

```diff
@@ -0,0 +1,339 @@
+        if in_pool_count > 0 then
+            pool_notification = colors.warning .. ' [' .. in_pool_count .. ']' .. colors.close
+        end
+        text = text .. "\n" .. get_indent(depth) .. pop.dropped_from.name .. "\n" .. get_indent(depth) .. ' >> ' .. item_colour .. item_identifier .. pop_name .. colors.close .. pool_notification
+        if pop.dropped_from.pops then
+            text = text .. generate_text(pop.dropped_from, key_items, items, depth + 1)
+        end
+    end
+
+    return text
+end
+
+EmpyPopTracker.add_to_chat = function(message)
+    if type(message) ~= 'string' then
+        error('add_to_chat requires the message arg to be a string')
+    end
+
+    windower.add_to_chat(8, message)
+end
+
+EmpyPopTracker.generate_info = function(nm, key_items, items)
+    local nm_type = type(nm)
+    if nm_type ~= 'table' then
+        error('generate_info requires the nm arg to be a table, but got ' .. nm_type .. ' instead')
+    end
+
+    local info = {
+        has_all_kis = true,
+        text = ""
+    }
+
+    if nm.pops then
+        for _, key_item_data in pairs(nm.pops) do
+            local has_pop_ki = owns_key_item(key_item_data.id, key_items)
+
+            if not has_pop_ki then
+                info.has_all_kis = false
+            end
+        end
+    end
+
+    info.text = generate_text(nm, key_items, items, 1)
+
+    return info
+end
+
+function find_nms(query)
+    local matching_nms = {}
+    local lower_query = query:lower()
+    for _, nm in pairs(nm_data) do
+        local result = string.match(nm.name:lower(), '(.*' .. lower_query .. '.*)')
+        if result then
+            table.insert(matching_nms, result)
+        end
+    end
+    return matching_nms
+end
+
+windower.register_event("addon command", function(command, ...)
+    if commands[command] then
+        commands[command](...)
+    else
+        commands.help()
+    end
+end)
+
+commands = {}
+
+commands.track = function(...)
+    local args = {...}
+    local nm_name = args[1]
+    local matching_nm_names = find_nms(nm_name)
+
+    if #matching_nm_names == 0 then
+        EmpyPopTracker.add_to_chat('Unable to find a NM using: "' .. nm_name .. '"')
+    elseif #matching_nm_names > 1 then
+        EmpyPopTracker.add_to_chat('"' .. nm_name .. '" matches ' .. #matching_nm_names .. ' NMs. Please be more explicit:')
+        for key, matching_file_name in pairs(matching_nm_names) do
+            EmpyPopTracker.add_to_chat('  Match ' .. key .. ': ' .. ucwords(matching_file_name))
+        end
+    else
+        active = true
+        EmpyPopTracker.add_to_chat("Now tracking: " .. ucwords(matching_nm_names[1]))
+        EmpyPopTracker.settings.tracking = matching_nm_names[1]
+        EmpyPopTracker.update()
+        commands.show()
+    end
+end
+commands.t = commands.track
+
+commands.hide = function()
+    active = false
+    EmpyPopTracker.text:visible(false)
+    EmpyPopTracker.settings.visible = false
+    EmpyPopTracker.settings:save()
```

Comment:
You should not need to modify the settings at all when doing this. It should adjust them and save accordingly. Does that not happen for you? Same in the function below.

Reply:
I originally expected this to just work like some of my other addons that use texts and settings. I just checked and it doesn't save it automatically :/

I had just assumed the texts visible setting wasn't saved like the others?

Reply:
Hm you are right, I just checked and visibility is not automatically saved. I wonder why I did it like that... I guess I did not consider visibility as a user based setting and instead wanted the addon to control that behavior. An argument I no longer agree with, but I think that's what influenced my decision back then.